airbyte-webapp/src/hooks/services/useSourceHook.tsx

```diff
@@ -180,15 +180,12 @@ const useDiscoverSchema = (
     } finally {
       setIsLoading(false);
     }
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [sourceId]);
+  }, [disableCache, service, sourceId]);
 
   useEffect(() => {
-    (async () => {
-      if (sourceId) {
-        await onDiscoverSchema();
-      }
-    })();
+    if (sourceId) {
+      onDiscoverSchema();
```

Comment:
How come this went from async to not async?

Reply:
The async wasn't affecting anything. Awaiting this within a function call that itself was not awaited meant it wasn't actually blocking anything.